modelplace_api/utils.py

```diff
@@ -17,6 +17,24 @@
     from ._rle_mask import encode_binary_mask, decode_coco_rle
 
 
+def is_numpy_array_equal(result: np.ndarray, gt: np.ndarray, error: float) -> bool:
+    '''
+    Applies element-wise comparison of two ndarrays and defines
+    whether the ratio of matched elements is eligible
+    '''
+    if result.size != gt.size:
+        raise RuntimeError('"result" and "gt" arrays must have equal size')
```

Comment:
Seems you also need to make sure ndtypes of the arrays are the same

Reply:
Added the check

Reply:
Also, I think better to compare `shape`s not `size`s as using `size`s won't raise an Exception however you'll get `ValueError: operands could not be broadcast together with shapes` in `np.equal` or `np.isclose`. You can check:

```python
import numpy as np

array = [1,2,3,4,5,6]

a = np.array(array).reshape((2,3))
b = np.array(array).reshape((3,2))

print(f"a.size==b.size is {a.size == b.size}")
print(f"a.shape==b.shape is {a.shape == b.shape}")

np.equal(a,b)
```